src/mds/MDSMap.h

```diff
@@ -275,6 +275,11 @@ class MDSMap {
   const std::string get_balancer() const { return balancer; }
   void set_balancer(std::string val) { balancer.assign(val); }
 
+  const std::bitset<MAX_MDS>& get_bal_rank_mask_bitset() const;
+  void set_bal_rank_mask(std::string val, std::string bin_string);
```

Comment:
```suggestion
  int set_bal_rank_mask(std::string val);
```

return error (to FSCommands.cc) if hex2bin or other validity checking fails

Reply:
In my current implementation, `hex2bin` validates and converts hex format to binary string, while set_bal_rank_mask only stores valid hex format.